packages/backend/tests/auth.spec.ts

```diff
@@ -0,0 +1,313 @@
+  let spyLogger: jest.SpyInstance;
+
+  beforeEach(() => {
+    mockRequest = {
+      headers: {},
+      body: {},
+    } as Request;
+
+    mockResponse = {
+      status: jest.fn(() => mockResponse),
+      json: jest.fn(),
+      send: jest.fn(),
+      end: jest.fn()
+    } as any as Response;
+
+    mockUser = {
+      userName: 'testuser#1234',
+      userId: '123456789',
+      createdAt: Date.now(),
+    };
+
+    mockToken = {
+      key: 123456,
+      createdAt: Date.now(),
+      userId: '123456789',
+    };
+
+    mockNext = jest.fn();
+    spyLogger = jest.spyOn(loggerFile, 'error');
+
+    jest.mock('../src/configuration/discord.ts')
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  
+  it('should log error if wrong authRequest is provided', async () => {
+    // no body
+    await authLoginRequest(mockRequest, mockResponse, mockNext);
+    expect(spyLogger.mock.calls.length).toBe(1);
+    expect(mockNext.mock.calls.length).toBe(1);
+    expect(mockNext.mock.calls[0][0]).toEqual(new ApiError(400, "\"username\" is required"));
+
+    // number as username (userid instead of username)
+    mockRequest.body.username = 12312312312312;
+
+    await authLoginRequest(mockRequest, mockResponse, mockNext);
+    expect(spyLogger.mock.calls.length).toBe(2);
+    expect(mockNext.mock.calls.length).toBe(2);
+    expect(mockNext.mock.calls[1][0]).toEqual(new ApiError(400, "\"username\" must be a string")); 
+    
+    // no token
+    mockRequest.body.username = mockUser.userName;
+
+    await authLoginRequest(mockRequest, mockResponse, mockNext);
+    expect(spyLogger.mock.calls.length).toBe(3);
+    expect(mockNext.mock.calls.length).toBe(3);
+    expect(mockNext.mock.calls[2][0]).toEqual(new ApiError(400, "\"token\" is required"));
+    
+    // token is too small
+    mockRequest.body.username = mockUser.userName;
+    mockRequest.body.token = "12312";
+
+    await authLoginRequest(mockRequest, mockResponse, mockNext);
+    expect(spyLogger.mock.calls.length).toBe(4);
+    expect(mockNext.mock.calls.length).toBe(4);
+    expect(mockNext.mock.calls[3][0]).toEqual(new ApiError(400, "\"token\" must be greater than 100000"));
+
+    // no successfull response
+    expect((mockResponse.status as jest.Mock).mock.calls.length).toBe(0);
+    expect((mockResponse.json as jest.Mock).mock.calls.length).toBe(0);
+  });
+
+  it('should log error if unkown user is provided', async () => {
+    mockRequest.body.username = "testuser2#123123";
+    mockRequest.body.token = mockToken.key;
+    mockingoose(User).toReturn(null, 'findOne');
+
+    await authLoginRequest(mockRequest, mockResponse, mockNext);
+
+    expect(mockNext.mock.calls.length).toBe(1);
+    expect(mockNext.mock.calls[0][0]).toEqual(new ApiError(404, `Nutzer ${mockRequest.body.username} nicht gefunden`));
+    expect(spyLogger.mock.calls.length).toBe(1);
+    expect(spyLogger.mock.calls[0][0]).toBe(`Nutzer ${mockRequest.body.username} nicht gefunden`);
```

Comment:
Auf Englisch wegen Einheitlichkeit?
```suggestion
    expect(mockNext.mock.calls[0][0]).toEqual(new ApiError(404, `User ${mockRequest.body.username} not found`));
    expect(spyLogger.mock.calls.length).toBe(1);
    expect(spyLogger.mock.calls[0][0]).toBe(`User ${mockRequest.body.username} not found`);
```

Reply:
Habe alles ins Englische übersetzt 👍